File: harshini/Kadane.py

def kadane(A):
	max_current = max_global = A[0]
	for i in range(len(A)-1):
		max_current = max(A[i],max_current+A[i])
		if max_current > max_global:
			max_global = max_current
			#print (max_current,max_global)

	return max_global

A = [-2, -3, 4, -1, -2, 1, 5, -3]  

print (kadane(A))

File: shruthisg/Kadane's Algorithm/Kadane.java

public class Kadane
{
    static int MaxSubarray(int a[])
    {
        int Max_current = a[0];
        int Max_Global = a[0];
        for(int i=1;i<a.length;i++)
        {
            if(Max_current < 0)
            Max_current = a[i];
            else
            Max_current += a[i];
            if(Max_current >= Max_Global )
            Max_Global = Max_current;
        }
       return Max_Global;
    }
    public static void mian()
    {
        int a[] = {-2,3,2,-1};
        System.out.println("Maximum sum of the sub array is : " +MaxSubarray(a));
    }
    
}
